esp/esp/web/views/main.py

```diff
@@ -225,20 +225,25 @@ def registration_redirect(request):
     ctxt = {}
     userrole = {}
     regperm = None
-    if user.isStudent():
-        userrole['name'] = 'Student'
-        userrole['base'] = 'learn'
-        userrole['reg'] = 'studentreg'
-        regperm = 'Student/Classes'
-    elif user.isTeacher():
+    if user.isTeacher():
         userrole['name'] = 'Teacher'
         userrole['base'] = 'teach'
         userrole['reg'] = 'teacherreg'
         regperm = 'Teacher/Classes'
+    elif user.isStudent():
+        userrole['name'] = 'Student'
+        userrole['base'] = 'learn'
+        userrole['reg'] = 'studentreg'
+        regperm = 'Student/Classes'
+
     ctxt['userrole'] = userrole
 
     if regperm:
-        progs = list(Permission.program_by_perm(user,regperm))
+        if user.isTeacher():
+            progs = list(Permission.program_by_perm(user,regperm))
+        else:
+            user_grade = user.getGrade()
+            progs = list(Permission.program_by_perm(user,regperm).filter(grade_min__lte=user_grade).filter(grade_max__gte=user_grade))
```

Comment:
It would be cleaner to have both of the filter arguments in the same filter call.